### PR TITLE
Mbst 13222 adding separate score reasons

### PR DESCRIPTION
**Pull request by MartinsIrbe**
Added changes to common.proto and content.proto to have two separate list for Priority score reasons.
Furthermore updated following classes for serialisation and deserialisation:
PrioritySerializer
ContentSerializationVisitor
ContentDeserializationVisitor
Updated tests:
PrioritySerializer

### CODE REVIEW

File: atlas-cassandra/src/main/java/org/atlasapi/content/ContentSerializationVisitor.java [hunk 1]
```diff
@@ -137,10 +137,15 @@ private Builder visitDescribed(Described content) {
             ContentProtos.Priority.Builder priorityBuilder = ContentProtos.Priority.newBuilder();
             Priority priority = content.getPriority();
             if (priority.getPriority() != null) {
-                priorityBuilder.addAllReasons(priority.getReasons());
-            }
-            if (priority.getPriority() != null) {
                 priorityBuilder.setScore(priority.getPriority());
+                if (priority.getReasons() != null) {
```
Comment (emilssolmanis) [inline atlas-cassandra/src/main/java/org/atlasapi/content/ContentSerializationVisitor.java]: Extract `priority.getReasons()` to a var, you're using it at least 6 times here.

File: atlas-cassandra/src/main/java/org/atlasapi/content/PrioritySerializer.java [hunk 2]
```diff
@@ -10,8 +10,15 @@
         CommonProtos.Priority.Builder builder = CommonProtos.Priority.newBuilder();
 
         if(source.getReasons() != null) {
-            for (String reason : source.getReasons()) {
-                builder.addReason(reason);
+            if (source.getReasons().getPositive() != null) {
```
Comment (emilssolmanis) [inline atlas-cassandra/src/main/java/org/atlasapi/content/PrioritySerializer.java]: Same as above, extract as var.

File: atlas-cassandra/src/main/protobuf/common.proto [hunk 3]
```diff
@@ -118,8 +118,9 @@ message Synopses {
 }
 
 message Priority {
-    repeated string reason = 1;
+    repeated string positive_reasons = 1;
```
Comment (emilssolmanis) [inline atlas-cassandra/src/main/protobuf/common.proto]: Reusing field IDs is a Very Bad Idea™ unless the fields are actually the same. Are you sure this is what you want? Doesn't seem like `reason` is the same field semantically as `positive_reasons`. For that matter, what happens to compatibility in this case?

File: atlas-cassandra/src/main/protobuf/content.proto [hunk 4]
```diff
@@ -119,7 +119,8 @@ message Restriction {
 
 message Priority {
     optional double score = 1;
-    repeated string reasons = 2;
+    repeated string positive_reasons = 2;
```
Comment (emilssolmanis) [inline atlas-cassandra/src/main/protobuf/content.proto]: Same as above.

File: atlas-cassandra/src/main/protobuf/common.proto (hunk 3 above)
Comment (MartinsIrbe) [inline atlas-cassandra/src/main/protobuf/common.proto]: I will update and mark `reason` as deprecated.

File: atlas-cassandra/src/main/protobuf/content.proto (hunk 4 above)
Comment (MartinsIrbe) [inline atlas-cassandra/src/main/protobuf/content.proto]: :+1:

File: atlas-cassandra/src/main/java/org/atlasapi/content/ContentSerializationVisitor.java (hunk 1 above)
Comment (MartinsIrbe) [inline atlas-cassandra/src/main/java/org/atlasapi/content/ContentSerializationVisitor.java]: :+1:

File: atlas-cassandra/src/main/java/org/atlasapi/content/PrioritySerializer.java (hunk 2 above)
Comment (MartinsIrbe) [inline atlas-cassandra/src/main/java/org/atlasapi/content/PrioritySerializer.java]: :+1:

**Comment by emilssolmanis** [comment]
:+1: :shipit:

**Pull request merged**